base/db/R/get.new.site.R

```diff
@@ -0,0 +1,121 @@
+##' Get new site info using provided site information
+##'
+##' @title Get New Site Info
+##' @param site a dataframe with site information including id, lat, lon, and time_zone.
+##' @param con Database connection object.
+##' @return a dataframe with new site information on lat, lon and time_zone
+##' @export
+##' @author Abhinav Pandey
+##' @importFrom digest digest
+##'
+##' @examples
+##' get.new.site(site=data.frame(id=1,lat=40.1,lon=-88.2, time_zone = "UTC"), con = NULL)
+
+get.new.site <- function(site, con = NULL) {
+    if (is.null(con)) {
+        PEcAn.logger::logger.debug("DB connection is closed. Trying to generate a new site ID or use pre-existing one.")
+        # No DB connection present. Generate a new ID using one of below steps:
+
+        if (is.null(site$id) | is.na(site$id)) {
+            if ((!is.null(site$lat) && !is.null(site$lon)) &&
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                site.id <- paste0(site$lat, "_", site$lon)
+                new.site <- data.frame(
+                    id = as.numeric(site.id),
```

Comment:
the as.numeric here seems like it's going to always return NA given how site.id is constructed